src/org/ods/orchestration/phases/DeployOdsComponent.groovy

```diff
@@ -58,15 +58,24 @@ class DeployOdsComponent {
                     applyTemplates(openShiftDir, deploymentMean)
 
                     def retries = project.environmentConfig?.openshiftRolloutTimeoutRetries ?: 10
+                    def podData = null
                     for (def i = 0; i < retries; i++) {
-                        def podData = os.checkForPodData(project.targetProject, deploymentMean.selector)
+                        podData = os.checkForPodData(project.targetProject, deploymentMean.selector)
                         if (podData) {
                             return podData
                         }
                         steps.echo("Could not find 'running' pod(s) with label '${deploymentMean.selector}' - waiting")
                         steps.sleep(12)
                     }
 
+                    if (podData == null) {
+                        throw new RuntimeException(
+                        "Error: Could not find 'running' pod(s) with label"
+                        + "'${deploymentMean.selector}'"
+                        )
+                    }
+
+                    // TODO: What if podData is empty?
                     // TODO: Once the orchestration pipeline can deal with multiple replicas,
```

Comment:
TODO's annotate unfinished work and should not appear in final code.
The case podData is empty should be handled above and the TODO can be removed.
With respect to the second TODO, I know it was there before, but please remove the TODO. Mentioning that currently we only support one instance is enough.